pgjdbc/src/main/java/org/postgresql/util/LazyCleaner.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2023, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+/* changes were made to move it into the org.postgresql.util package
+ *
+ * Copyright 2022 Juan Lopes
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.postgresql.util;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.lang.ref.PhantomReference;
+import java.lang.ref.ReferenceQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class LazyCleaner {
+  private static final Logger LOGGER = Logger.getLogger(LazyCleaner.class.getName());
+
+  public interface Cleanable {
+    void clean();
+  }
+
+  public interface CleaningAction<T extends Throwable> {
+    void onClean(boolean leak) throws T;
+  }
+
+  private final ReferenceQueue<Object> queue = new ReferenceQueue<Object>();
+  private final long threadTtl;
+  private final ThreadFactory threadFactory;
+  private boolean threadRunning = false;
+  private int watchedCount = 0;
+  private @Nullable Node first;
+  private @Nullable Cleanable keepAliveCleanable;
+
+  public LazyCleaner(long threadTtl, final String threadName) {
+    this(threadTtl, new ThreadFactory() {
+      public Thread newThread(Runnable runnable) {
+        Thread thread = new Thread(runnable, threadName);
+        thread.setDaemon(true);
+        return thread;
+      }
+    });
+  }
+
+  public LazyCleaner(long threadTtl, ThreadFactory threadFactory) {
+    this.threadTtl = threadTtl;
+    this.threadFactory = threadFactory;
+  }
+
+  public synchronized LazyCleaner setKeepThreadAlive(boolean alive) {
```

Comment:
I'm trying to figure out what this does? It is never actually called. Do we need it ?

Reply:
I do not think we need this, so I'm fine if we remove it for simplicity

Reply:
+1

Reply:
removed